Sources/NIOHTTP2/HTTP2CommonInboundStreamMultiplexer.swift

```diff
@@ -304,10 +311,30 @@ extension HTTP2CommonInboundStreamMultiplexer {
             inboundStreamStateInitializer: .excludesStreamID(nil)
         )
         self.pendingStreams[channel.channelID] = channel
-        channel.configure(initializer: streamStateInitializer, userPromise: promise)
+
+        let anyInitializer: NIOChannelInitializerWithOutput<any Sendable> = { channel in
+            streamStateInitializer(channel).map { return $0 }
+        }
+
+        let anyPromise: EventLoopPromise<Any>?
+        if let promise = promise {
+            anyPromise = channel.baseChannel.eventLoop.makePromise(of: Any.self)
+            anyPromise?.futureResult.whenComplete { result in
+                switch result {
+                case .success(let any):
+                    promise.succeed(any as! Output)
```

Comment:
Can we add this comment here
```suggestion
                    // The cast through any here is unfortunate but the only way to make this work right now
                    // since the HTTP2ChildChannel and the multiplexer is not generic over the output of the initializer.
                    promise.succeed(any as! Output)
```